packages/Chem/src/OCL-service/OCL-worker.ts

```diff
@@ -4,26 +4,31 @@ import * as OCL from 'openchemlib/full';
 
 type OCLWorkerReturnType = {res: {[key: string]: Array<number>}, errors: string[]};
 
-onmessage = ({data: {op, data, argList}}) => {
+onmessage = ({data: {op, data, argList, notationType}}) => {
   switch (op) {
   case OCLServiceCall.CHEM_PROPERTIES:
-    postMessage(getChemProperties(data, argList));
+    postMessage(getChemProperties(data, argList, notationType));
     break;
+  case OCLServiceCall.DRUG_LIKENESS:
+    postMessage(getDrugLikeliness(data, notationType));
   default:
-    postMessage(getToxRisks(data, argList));
+    postMessage(getToxRisks(data, argList, notationType));
     break;
   }
 };
 
+function isSmiles(notationType: string): boolean {
+  return notationType === 'smiles';
+}
 
-function getChemProperties(molList: Array<string>, propList: string[]): OCLWorkerReturnType {
+function getChemProperties(molList: Array<string>, propList: string[], notationType: string): OCLWorkerReturnType {
   const res: {[key: string]: Array<number>} = {};
   const errors: string[] = [];
   for (const propName of propList)
-    res[propName] = new Array(molList.length).fill(NaN);
+    res[propName] = new Array(molList.length).fill(null);
```

Comment:
why do we fill it with nulls?

Reply:
for some molecules operation can possibly cause error, due to it being malformed or whatnot. if its filled with NaN, it will show up as NaN in the cell, null will result in cell being empty, which is what we want